### ISSUE TITLE
suffering from high cpu usage (100%) when running with Pypy3.5 v6.0.0 

### ISSUE THREAD

**Issue by ArchieMeng** [python-telegram-bot/python-telegram-bot#1194]
### Steps to reproduce
1. install pypy3 v6.0.0

2. run bot with this pypy(exp. https://github.com/ArchieMeng/telegram_calc_bot, this is my bot)

### Expected behaviour
Low CPU usage which should not be any high than 5% when waiting for message.  

### Actual behaviour
high CPU usage which is almost 100%, seems like take up the whole CPU time slice.

### Configuration
**Operating System:**
Ubuntu 16.04 Server
Linux ubuntu 4.10.17-041017-generic #201705201051 SMP Sat May 20 14:53:33 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Version of Python, python-telegram-bot & dependencies:**
Python version: Python 3.5.3 (fdd60ed87e94, Apr 24 2018, 06:10:04) [PyPy 6.0.0 with GCC 6.2.0 20160901]

***library dependencies:***
>certifi==2018.8.13
cffi==1.11.5
future==0.16.0
greenlet==0.4.13
python-telegram-bot==10.1.0
readline==6.2.4.1
six==1.11.0

### Logs
Insert logs here (if necessary)
no outputs, but a screenshot about what had happened.
![2018-08-23 11-52-42](https://user-images.githubusercontent.com/12139431/44504105-7a49db80-a6cc-11e8-96a1-cefc0bb6a466.png)

**Comment by jsmnbom** [comment]
Does this only happen on pypy?

**Comment by ArchieMeng** [comment]
Yes!
It is currently running very well under Python3.6 . This situation only happens when running with pypy3.5 (6.0.0, latest source build.), By the way, all of my bot suffer from this.
And, any methods for me to debug on this?

**Comment by ArchieMeng** [comment]
To be more accurate, CPU usage goes high after execute:
```
self._init_thread(self._start_polling, "updater", poll_interval, timeout,
                                  read_latency, bootstrap_retries, clean, allowed_updates)
```
in updater.start_polling().
Oh, BTW, I am running this bot with "workers" set to 4.

**Comment by jsmnbom** [comment]
Hmm, I am personally not very familiar with how pypy works... But it seems like it could be stuck in a loop in one of those threads somewhere... Could you try using a debugger, pausing the threads and seeing where they are executing/getting stuck (if that's what happening)?

**Comment by ArchieMeng** [comment]
@jsmnbom Yes, I find it struggling to run conn.urlopen(xxx), in which conn is a HTTPSConnectionPool from telegram/vendor/ptb_urllib3/urllib3/poolmanager.py . And this is the critical part that make cpu usage go that high.

**Comment by jsmnbom** [comment]
This seems like it might be an issue with upstream (or possibly our own vendored) urllib3 then :/

**Comment by tsnoam** [comment]
I've managed to easily recreate this issue using `echobot2`
1. The issue is not limited to pypy3-6.0.0. It happens with pypy3-5.7.0.
2. The issue has nothing to do with urllib3 (or the vendored lib for that matter) as it goes back to v4.0.1 (haven't tested prior to that version).

Will keep on investigating.
